pkg/externalpipeline/externalpipeline.go

```diff
@@ -0,0 +1,165 @@
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/openshift/insights-operator/pkg/config"
+)
+
+type ReportsCache struct {
+	Configuration config.SmartProxy
+	client        *http.Client
+	authorizer    Authorizer
+	clusterInfo   ClusterVersionInfo
+	LastReport    types.SmartProxyReport
+}
+
+type InsightsReportResponse struct {
+	Report types.SmartProxyReport `json:"report"`
+	Status string                 `json:"status"`
+}
+
+type ClusterVersionInfo interface {
+	ClusterVersion() *configv1.ClusterVersion
+}
+
+type Authorizer interface {
+	Authorize(req *http.Request) error
+	NewSystemOrConfiguredProxy() func(*http.Request) (*url.URL, error)
+}
+
+var (
+	// ErrWaitingForVersion contains an error when the clusver version is not loaded yet
+	ErrWaitingForVersion = fmt.Errorf("waiting for the cluster version to be loaded")
+
+	// insightsStatus contains a metric with the latest report information
+	insightsStatus = metrics.NewGaugeVec(&metrics.GaugeOpts{
+		Namespace: "health",
+		Subsystem: "statuses",
+		Name:      "insights",
+		Help:      "Information about the cluster health status as detected by Insights tooling.",
+	}, []string{"metric"})
+)
+
+func New(c config.SmartProxy, authorizer Authorizer, clusterInfo ClusterVersionInfo) *ReportsCache {
+	return &ReportsCache{
+		Configuration: c,
+		client:        &http.Client{},
+		authorizer:    authorizer,
+		clusterInfo:   clusterInfo,
+	}
+}
+
+func (r *ReportsCache) PullSmartProxy() error {
+	klog.Info("Pulling report from smart-proxy")
+	cv := r.clusterInfo.ClusterVersion()
+	if cv == nil {
+		klog.Warning(ErrWaitingForVersion)
+		return ErrWaitingForVersion
+	}
+
+	endpoint := fmt.Sprintf(r.Configuration.Endpoint, cv.Spec.ClusterID)
+	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+
+	klog.Infof("Retrieving report for cluster: %s", cv.Spec.ClusterID)
+	klog.Infof("Endpoint: %s", endpoint)
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	if req.Header == nil {
+		req.Header = make(http.Header)
+	}
+	req.Header.Set("User-Agent", fmt.Sprintf("insights-operator/%s cluster/%s", version.Get().GitCommit, cv.Spec.ClusterID))
+
+	if err := r.authorizer.Authorize(req); err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	klog.Infof("Headers: %v\n", req.Header)
+
+	response, err := r.client.Do(req)
+	if err != nil {
+		klog.Errorf("Unable to retrieve latest report for %s: %v", cv.Spec.ClusterID, err)
+		return err
+	}
+
+	if response.StatusCode == http.StatusOK {
+		body, err := ioutil.ReadAll(response.Body)
+
+		if err != nil {
+			klog.Error("The response from Smart Proxy cannot be read")
+			return err
+		}
+
+		reportResponse := InsightsReportResponse{}
+		if err = json.Unmarshal(body, &reportResponse); err != nil {
+			klog.Error("The report response cannot be parsed")
+		} else {
+			updateInsightsMetrics(reportResponse.Report)
+			r.LastReport = reportResponse.Report
+		}
+	}
+
```

Comment:
prob. log info about StatusCode that is different from StatusOK?

Reply:
Agree. You will see it on the next commit

Reply:
It is now handled by `insightsclient` package